objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/NetworkStreamRetriever.java

```diff
@@ -62,6 +63,15 @@ public void setupDataRetrieval(Map<String, ?> dataSourceConfig, ObjectRecognitio
         }
         if (dataSourceConfig.get("camera") instanceof String){
             camera = (String) dataSourceConfig.get("camera");
+            try {
+                URI httpCheck = new URI(camera);
+                if (!(httpCheck.getScheme().equals("http")) && !(httpCheck.getScheme().equals("https"))) {
```

Comment:
On reflection, is this check enough?  Suppose we got Brett's URL (```https://.../playlist.m3u8```).  That would not be determined to be a push protocol, and I think it wouldn't fix the underlying problem.

Maybe we check the protocol in use OR if we're being directed there from some mpeg playlist (~~httpCheck~~ ```pushCheck.getPath().endsWith("m3u") || pushCheck.getPath().endsWith("m3u8")```).

There probably needs to be something more general, but that'll do for now.